parts/k8s/kubernetescustomscript.sh

```diff
@@ -167,10 +160,6 @@ if $FULL_INSTALL_REQUIRED; then
         # mitigation for bug https://bugs.launchpad.net/ubuntu/+source/linux/+bug/1676635
         echo 2dd1ce17-079e-403c-b352-a1921ee207ee > /sys/bus/vmbus/drivers/hv_util/unbind
         sed -i "13i\echo 2dd1ce17-079e-403c-b352-a1921ee207ee > /sys/bus/vmbus/drivers/hv_util/unbind\n" /etc/rc.local
-
-        wait_for_apt_locks
-        retrycmd_if_failure 120 5 25 apt-mark unhold walinuxagent || exit $ERR_RELEASE_HOLD_WALINUXAGENT
```

Comment:
we used to have a separate exit code for release and hold, can we have two different functions for hold and release? Since `wait_for_apt_locks` is different for hold and release that way we can keep the `wait_for_apt_locks` commands in the hold function and not have them in the release function. So something like: 

```
holdWALinuxAgent() {
    if [[ $OS == $UBUNTU_OS_NAME ]]; then
        wait_for_apt_locks
        retrycmd_if_failure 120 5 25 apt-mark hold walinuxagent || exit $ERR_HOLD_WALINUXAGENT
        wait_for_apt_locks
    fi
}

releaseWALinuxAgent() {
    if [[ $OS == $UBUNTU_OS_NAME ]]; then
        retrycmd_if_failure 120 5 25 apt-mark unhold walinuxagent || exit $ERR_RELEASE_HOLD_WALINUXAGENT
    fi
}
```

Reply:
Just trying to keep the payload surface area small by having a single func that takes an arg for "hold" or "unhold". For error conditions the func returns a distinct error code based on the invocation context.